### PR TITLE
fix: use appropriate copy for FCFS vs Lottery in Application Confirmation screen

### PR DESCRIPTION
**Pull request by jaredcwhite**
# Pull Request Template

## Issue Overview

This PR addresses #2510 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

In QA testing [it was discovered](https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/bloom-housing/bloom/2510#issuecomment-1109969397) the original PR #2558 only included copy for First Come First Serve listings. This updates the application confirmation screen so that Lottery listings will also show the correct copy.

## How Can This Be Tested/Reviewed?

The [deploy preview](https://deploy-preview-2702--dev-bloom.netlify.app/) has an outdated DB seed so the two listing types will need to be tested locally.

Fill out an application for both types and see if the final confirmation screen shows either:

* Eligibile applicants will be contacted by on a **first come first serve** basis until vacancies are filled
* The lottery will be held on **(DATE)**. Eligible applicants will be contacted by the agent in lottery rank order until vacancies are filled

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: sites/public/pages/applications/review/confirmation.tsx [hunk 1]
```diff
@@ -26,6 +28,20 @@ const ApplicationConfirmation = () => {
 
   const imageUrl = imageUrlFromListing(listing, parseInt(process.env.listingPhotoSize))
 
+  const reviewOrder = useMemo(() => {
+    if (listing) {
+      if (listing.reviewOrderType == ListingReviewOrder.lottery) {
```
Comment (emilyjablonski) [inline sites/public/pages/applications/review/confirmation.tsx:33]: A listing can be saved as having a review order type of lottery and no application due date, so the existence of that field should also be checked so we can't end up in a broken string state.

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
Check out the comment above around checking for the string fields before using them

**Comment by jaredcwhite** [comment]
> Check out the comment above around checking for the string fields before using them

Good to know! I think this will effect the emails that go out as well so I'll also look that over again.

**Comment by jaredcwhite** [comment]
All right, I should have that handled now. I'm not sure if published lottery listings do happen to go out without a due date present, but if that's an issue probably needs to get handled on the backend validation side of things.

**Review by emilyjablonski — APPROVED** [review]
LGTM!

**Pull request merged**